### PR TITLE
Introduce java.lang.Class.serializerOrNull() that is equivalent of KClass.serializerOrNull(), but attempts not to instantiate KClass when possible

### PR DESCRIPTION
**Pull request by qwwdfsad**

### CODE REVIEW

File: core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt [hunk 1]
```diff
@@ -74,6 +74,19 @@ public fun SerializersModule.serializer(type: Type): KSerializer<Any> =
...
+ * Retrieves a [KSerializer] for the given [java.lang.Class] instance of the Kotlin type or returns `null` if none is found.
+ * The given class must be annotated with [Serializable] or be one of the built-in types.
+ *
+ * The API avoids instantiation of the correspnding [KClass][Class.kotlin] on the best-effort basis.
+ *
+ * This is a [Class] counterpart of [KClass.serializerOrNull] with all the restrictions the original function implies,
+ * including the general recommendation to avoid uses of this API.
+ */
+@InternalSerializationApi
+public fun <T : Any> Class<T>.serializerOrNull(): KSerializer<T>? =
```
Comment (qwwdfsad) [inline core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt:87]: Keep in mind that we already have `serializerOrNull` on `Type` that does exactly this and a bit of extra (e.g. checks for generics), I've just found out that making a separate overload is much easier from the maintenance standpoint than avoiding `KClass` instantiation on each path of `serializerOrNull(Type)`

File: core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt [hunk 2]
```diff
@@ -74,6 +74,19 @@ public fun SerializersModule.serializer(type: Type): KSerializer<Any> =
 public fun SerializersModule.serializerOrNull(type: Type): KSerializer<Any>? =
     serializerByJavaTypeImpl(type, failOnMissingTypeArgSerializer = false)
 
+/**
+ * Retrieves a [KSerializer] for the given [java.lang.Class] instance of the Kotlin type or returns `null` if none is found.
+ * The given class must be annotated with [Serializable] or be one of the built-in types.
+ *
+ * The API avoids instantiation of the correspnding [KClass][Class.kotlin] on the best-effort basis.
```
Comment (ihostage) [inline core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt:81]: typo
```suggestion
 * The API avoids instantiation of the corresponding [KClass][Class.kotlin] on the best-effort basis.
```

File: core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt (hunk 1 above)
Comment (sandwwraith) [reply core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt:87]: I'm still not sure we should have a separate overload, especially `@Internal` one. Maybe just rewrite `is Class<*>` branch in the implementation?

Comment (qwwdfsad) [reply core/jvmMain/src/kotlinx/serialization/SerializersJvm.kt:87]: That's a great point, I'll give it a try

**Comment by qwwdfsad** [comment]
@sandwwraith  PTAL

File: core/commonMain/src/kotlinx/serialization/Serializers.kt [hunk 3]
```diff
@@ -137,7 +136,7 @@ internal fun <T : Any> SerializersModule.reflectiveOrContextual(kClass: KClass<T
  *
  * The recommended way to retrieve the serializer is inline [serializer] function and [`serializer(KType)`][serializer]
  *
- * This API is not guaranteed to work consistent across different platforms or
+ * This API is not guaranteed to work consistently across different platforms or
```
Comment (sandwwraith) [inline core/commonMain/src/kotlinx/serialization/Serializers.kt:139]: pls fix it in the docs for `KClass.serializer()` too, then

**Review by sandwwraith — APPROVED** [review]

File: core/commonMain/src/kotlinx/serialization/Serializers.kt (hunk 3 above)
Comment (qwwdfsad) [reply core/commonMain/src/kotlinx/serialization/Serializers.kt:139]: Already done :)

**Pull request merged**